ambari-metrics-timelineservice/src/main/java/org/apache/ambari/metrics/core/timeline/discovery/TimelineMetricMetadataManager.java

```diff
@@ -833,6 +855,9 @@ public boolean isTransientMetric(String metricName, String appId) {
    * Run TimelineMetricMetadataSync once
    */
   public void forceMetricsMetadataSync() {
+    if (metricMetadataSync == null) {
```

Comment:
Does this need to be atomic and volatile?

Reply:
Used only in migration and tests. So we are good.